### PR TITLE
Add datasets

### PR DESCRIPTION
**Pull request by oblute**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
-  [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by oblute** [comment]
@Tata17 agree to maintain?

**Comment by oblute** [comment]
@conda-forge/staged-recipes this is ready for review

File: recipes/datasets/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  imports:
+    - datasets
+    - datasets.commands
+    - datasets.utils
+
+about:
+  home: https://github.com/huggingface/datasets
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE
```
Comment (synapticarbors) [inline recipes/datasets/meta.yaml:44]: I opened a PR upstream to fix having to manually package the license file (https://github.com/huggingface/datasets/pull/1007). Please add the following comment as a visual reminder to check the status of whether the license file is in the source distribution:

```yaml
# License file manually packaged. See: https://github.com/huggingface/datasets/pull/1007
license_file: LICENSE
...
```

File: recipes/datasets/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3,<3.7
```
Comment (synapticarbors) [inline recipes/datasets/meta.yaml:20]: Do we know that this won't run on `python >=3.7` and that it's not just that the metadata in the upstream `setup.py` just hasn't been updated? I can see issues posted where people are using python 3.7 and higher and there are no comments that it isn't supported.

Comment (oblute) [reply recipes/datasets/meta.yaml:20]: Testing it

**Comment by oblute** [comment]
@synapticarbors good to go

**Pull request merged**